src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -95,6 +95,244 @@ public static FunctionPointerMethodSymbol CreateFromSource(FunctionPointerTypeSy
         public static FunctionPointerMethodSymbol CreateFromMetadata(CallingConvention callingConvention, ImmutableArray<ParamInfo<TypeSymbol>> retAndParamTypes)
             => new FunctionPointerMethodSymbol(callingConvention, retAndParamTypes);
 
+        public FunctionPointerMethodSymbol SubstiteParameterSymbols(
+            TypeWithAnnotations substitutedReturnType,
+            ImmutableArray<TypeWithAnnotations> substitutedParameterTypes,
+            ImmutableArray<CustomModifier> refCustomModifiers,
+            ImmutableArray<ImmutableArray<CustomModifier>> paramRefCustomModifiers)
+            => new FunctionPointerMethodSymbol(
+                this.CallingConvention,
+                this.RefKind,
+                substitutedReturnType,
+                refCustomModifiers.IsDefault ? this.RefCustomModifiers : refCustomModifiers,
+                this.Parameters,
+                substitutedParameterTypes,
+                paramRefCustomModifiers);
+
+        internal FunctionPointerMethodSymbol MergeEquivalentTypes(FunctionPointerMethodSymbol signature, VarianceKind variance)
+        {
+            Debug.Assert(RefKind == signature.RefKind);
+            var returnVariance = RefKind == RefKind.None ? variance : VarianceKind.None;
+            var mergedReturnType = ReturnTypeWithAnnotations.MergeEquivalentTypes(signature.ReturnTypeWithAnnotations, returnVariance);
+            var mergedReturnRefCustomMods = mergeCustomModifiers(RefCustomModifiers, signature.RefCustomModifiers, returnVariance);
```

Comment:
> var mergedReturnRefCustomMods = mergeCustomModifiers(RefCustomModifiers, signature.RefCustomModifiers, returnVariance); [](start = 11, length = 120)

This doesn't feel right. Custom modifiers are not part of the language and I cannot even think what would it mean to merge them, we don't even make sure they are compatible, etc. Unless we already have an example of doing this, for example during previous call, we shouldn't be doing this here. Follow what ```ReturnTypeWithAnnotations.MergeEquivalentTypes``` would do for custom modifiers